optimize/apis-tools/optimize-api/tutorial.md

```diff
@@ -0,0 +1,167 @@
+:::
+
+1. To set up your credentials, create an `.env` file which will be protected by the `.gitignore` file. These keys will be consumed by the `auth.js` file to execute the OAuth protocol, and should be saved when you generate your client credentials in [prerequisites](#prerequisites).
+2. Examine the existing `.env.example` file for an example of how your `.env` file should look upon completion. You will need to add your `COMPONENTS_CLIENT_ID`, `COMPONENTS_CLIENT_SECRET`, `OPTIMIZE_BASE_URL`, and `OPTIMIZE_AUDIENCE`, which is `optimize.camunda.io` in a Camunda 8 SaaS environment. Do not place your credentials in the `.env.example` file, as this example file is not protected by the `.gitignore`.
+
+:::note
+
+In this tutorial, we will execute arguments to list existing dashboard IDs and delete a dashboard. You can examine the framework for processing these arguments in the `cli.js` file before getting started.
+
+:::
+
+## GET a list of existing dashboard IDs
+
+First, let's script an API call to list our existing dashboard IDs.
+
+To do this, take the following steps:
+
+1. Examine the function `async function listDashboards([collectionId])` at the top of the `optimize.js` file. This is where you will script out your API call.
+2. Within the function, you must first apply an access token for this request:
+
+```
+const accessToken = await getAccessToken("components", optimizeAudience);
+```
+
+3. Using your generated client credentials from [prerequisites](#prerequisites), capture your Optimize audience and base URL. Ensure these credentials are added to your `.env` file, and retrieve the values from the process environment:
+
+```
+const optimizeAudience = process.env.OPTIMIZE_AUDIENCE;
+const optimizeApiUrl = process.env.OPTIMIZE_BASE_URL;
+```
+
+4. Script the API endpoint to list the dashboard IDs of a particular collection:
+
+```
+const url = `${optimizeApiUrl}/api/public/dashboard?collectionId=${collectionId}`;
+```
+
+5. Configure your GET request to the appropriate endpoint, including an authorization header based on the previously acquired `accessToken`:
+
+```
+  const options = {
+    method: "GET",
+    url,
+    headers: {
+      Accept: "application/json",
+      Authorization: `Bearer ${accessToken}`
+    }
+  };
+```
+
+6. Call the collection's endpoint, process the results from the API call, emit the dashboard IDs to output, and emit an error message from the server if necessary:
+
+```
+  try {
+    const response = await axios(options);
+    const results = response.data;
+
+    results.forEach(x => console.log(`ID: ${x.id}`));
+  } catch (error) {
+    // Emit an error from the server.
+    console.error(error.message);
+  }
+```
+
+7. In your terminal, run `node cli.js optimize list` for a list of your existing dashboard IDs. If you have any existing dashboards within a collection, you will see an output similar to the following:
```

Comment:
I needed to provide a collectionId here. Is that intentional? 

If I don't interpret step 4 correctly, I miss this and don't get an error or anything to guide me forward.

Also, I didn't see anything about how to find the collectionId. That should be added.

Reply:
Yes! This would be a good place to note that, because I don't think @toco-cam anticipated that either. Will add into a commit tomorrow 👍